### ISSUE TITLE
[Bug]: It's not clear to users that their calendar is failing/ how to reimport it

### ISSUE THREAD

**Issue by r-ferrier** [geeksforsocialchange/PlaceCal#2305]
This issue is about the error state for a calendar that has failed import

## Story
Maybe we could improve the UI around this again - some more informative text? A clearer definition of what the problem is and how it can be fixed by the admin user.

## Further Context
https://www.notion.so/gfsc/Kim-s-timeline-for-completion-12-3-24-d0e8ef7a03c44ba08e56d4477e3fea5d?pvs=4#bb7290e3d3e449df95bcd0cac435d34f

Blocked by #2304 which is the loading state for this UI component.

## Acceptance Criteria
- [x] Once saved, the status of the calendar is visible at the top of the page
- [x] I can see details about the status of my calendar near the lozenge
- [x] The details about status include lozenge, date updated, description (including notices & any error details)
- [x] Placement & design of error description annotices TBC... we might want to look at formatting notices as a separate issue
- [x] Once saved, I can always see re-import button & trigger re-import
- [x] I **cannot** schedule a timed import for my calendar

If the calendar has:
- [ ] Not finished importing - `in_queue` or `in_worker` (loading) lozenge & notice to check back in 10 minutes
- [ ] Event problem - `idle` (success) lozenge & I see notices for the failing events & a message suggesting I fix then re-import
- [ ] Invalid response - `error` (error) lozenge & notice to (TBC)
- [ ] Inaccessible feed - `bad_source` (error) lozenge & notice to (TBC)
- [ ] ActiveRecord error - `error` (error) lozenge & notice to (TBC)
- [ ] #2399

## Plan
- [x] Change status lozenge copy from (idle --> success, in_queue --> loading, in_worker --> loading, error --> error, bad_source --> error)
- [x] Button copy from "Queue for import now" to Re-import calendar
- [ ] more tbc...

Placement of status block:
![image](https://github.com/geeksforsocialchange/PlaceCal/assets/861172/3ff14682-ab03-4c6d-a24c-2ad24bd57124)

Status block changes:
![image](https://github.com/geeksforsocialchange/PlaceCal/assets/861172/20b5d49b-219f-4695-8f8a-8f85e7a52b07)

**Comment by ivan-kocienski-gfsc** [comment]
TODO: 

How can a calendar fail? What actions can be taken by the user?

**Comment by r-ferrier** [comment]
@katjam & @ivan-kocienski-gfsc to discuss the approach to this and work out AC

**Comment by katjam** [comment]
@kimadactyl Do you think we can drop the import from future date feature?

**Comment by kimadactyl** [comment]
don't see why not. i never used it and can't remember why it was added - maybe something to do with debugging?

**Comment by ivan-kocienski-gfsc** [comment]
```ruby

  rescue_from UnsupportedFeed do |exception|
    report_error exception, 'Calendar URL is not supported'
  end

  rescue_from InaccessibleFeed do |exception|
    report_bad_source_error exception.message
  end

  rescue_from InvalidResponse do |exception|
    report_error exception, 'Calendar URL returned un-parsable data'
  end

  rescue_from ActiveRecord::ActiveRecordError do |exception|
    raise exception if !Rails.env.production? && @silence_db_exceptions == false

    report_error exception, 'Internal database error'
  end
```

**Comment by katjam** [comment]
@honor-gfsc @kimadactyl How do these acceptance criteria look?

We need to firm up on copy too - but that can be secondary.

**Comment by kimadactyl** [comment]
Struggling to parse this but I trust you!

small note maybe we should put "edit calendar" and "view events" as subnavigation tabs? just an idea?

**Comment by ivan-kocienski-gfsc** [comment]
Is the notices mechanism useful to end users? On my current production snapshot I have the following:

```
calendar 16:
 notices:
  ["No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side",
   "No place or address could be created or found for the event location: Alternates between Hulme and Moss Side"],

calendar 93:
 notices:
  ["No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   (143 of these)

calendar 94:
 notices:
  ["Could not determine where this event is. A default location is set but the importer is set to ignore this. Add an address to the location field of the source calendar, or choose another import strategy with a default location.",
   "Could not determine where this event is. A default location is set but the importer is set to ignore this. Add an address to the location field of the source calendar, or choose another import strategy with a default location.",
   "Could not determine where this event is. A default location is set but the importer is set to ignore this. Add an address to the location field of the source calendar, or choose another import strategy with a default location.",
   "Could not determine where this event is. A default location is set but the importer is set to ignore this. Add an address to the location field of the source calendar, or choose another import strategy with a default location."],

calendar 109:
 notices:
  ["No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   "No place or address could be created or found for the event location: ",
   (53 times)

calendar 150:
 notices: ["No place or address could be created or found for the event location: "],

calendar 151:
 notices: ["No place or address could be created or found for the event location: Zoom meeting"],

```

It feels overwhelming when you have 150 indecipherable messages that have no actionable or identifiable information.

Maybe:
* we could find a way to just count all the messages that are the same and show them just the one message with a count?
* actually store something unique about each message?

**Comment by ivan-kocienski-gfsc** [comment]
So a couple of points coming from the discussion:

quick fix: De-dupe the messages and provide a count

longer fix: Log the title of the event into the message

**Comment by kimadactyl** [comment]
to me its a total nonissue to show the same message a bunch of times, if its any more than a 1 liner i wouldnt bother

i think the more important thing is its easy for the person to have a direction how to fix it - ideally it would link to their source event

**Comment by ivan-kocienski-gfsc** [comment]
If a user re-queues a calendar do we (should we?) clear the notices?

**Comment by katjam** [comment]
> If a user re-queues a calendar do we (should we?) clear the notices?

Yes. They will reappear if the same fails happen.
